segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -246,6 +249,80 @@ private static void validateContainerIds(Map<Integer, DebugStreamSegmentContaine
+     * are passed as parameters.
+     * @param storage                    A {@link Storage} instance where segments are stored.
+     * @param containerId                A Container Id to get the name of the metadata segment.
+     * @param backUpMetadataSegmentName  A name to the back up metadata segment.
+     * @param backUpAttributeSegmentName A name to the back attribute segment.
+     * @param executorService            A thread pool for execution.
+     * @return                           CompletableFuture which when completed will have the segments' contents copied to another segments.
+     */
+    public static CompletableFuture<Void> backUpMetadataAndAttributeSegments(Storage storage, int containerId,
+                                                                             String backUpMetadataSegmentName,
+                                                                             String backUpAttributeSegmentName,
+                                                                             ExecutorService executorService) {
+        Preconditions.checkNotNull(storage);
+        String metadataSegmentName = NameUtils.getMetadataSegmentName(containerId);
+        String attributeSegmentName = NameUtils.getAttributeSegmentName(metadataSegmentName);
+        return copySegment(storage, metadataSegmentName, backUpMetadataSegmentName, executorService)
+                        .thenAcceptAsync(x -> copySegment(storage, attributeSegmentName, backUpAttributeSegmentName,
+                                executorService));
+    }
+
+    /**
+     * Given the back up of metadata segments, this method iterates through all segments in it and gets their attributes
+     * and updates it in the container metadata segment of the given {@link DebugStreamSegmentContainer} instance.
+     * @param metadataSegments  A map of back of metadata segments along with their container Ids.
+     * @param containersMap     A map of {@link DebugStreamSegmentContainer} instances with their container Ids.
+     * @param executorService   A thread pool for execution.
+     * @throws Exception        If an exception occurred. This could be one of the following:
+     *                              * TimeoutException:     If the calls for computation(used in the method)
+     *                                                      didn't complete in time.
+     *                              * IOException     :     If a general IO exception occurred.
+     */
+    public static void updateCoreAttributes(Map<Integer, String> metadataSegments,
+                                            Map<Integer, DebugStreamSegmentContainer> containersMap,
+                                            ExecutorService executorService) throws Exception {
+        val args = IteratorArgs.builder().fetchTimeout(TIMEOUT).build();
+        SegmentToContainerMapper segToConMapper = new SegmentToContainerMapper(containersMap.size());
```

Comment:
You assume that your input is correct. Can you please do the following validations:
- metadataSegments.size == containersMap.size
- All container ids are accounted for

Reply:
Can there be a situation like this:
- We are given a storage instance, in pravega was running with just two containers.
- Migrate the storage instance to new pravega with 4 containers.
So, with the given storage instance with two container metadata segments, we get their back up copies. Now we iterate through 2 back up copies and update the segments in 4 containers. So, a precheck will fail in this case.

Reply:
No we cannot have that. We do not currently support changing the number of Segment Containers in a cluster (we may do this - #1644).